### PR TITLE
Jr/dev calculate bingo

### PR DESCRIPTION
**Pull request by nojronatron**
Designed and implemented bingo checking functions with Jest test cases and data for testing. Refactored several times to integrate into the Gameboard. BINGO result rendering is totally temporary and can be changed i.e. Gameboard still shows board while a Bingo is announced with future Bingo-party effects etc.

- [X] Develop components that can be called to test for Bingo.
- [X] Develop Jest test cases for each bingo checking component.
- [X] Integrate bingo checking into GameBoard.
- [X] Set placeholder logic to indicate Bingo when it happens.
- [X] All tests are passing.

### CODE REVIEW

**Review by schaferyan — COMMENTED** [review]
Looks good and it works! My only question is about the test files. I may be missing something but it looks like you could just parse the json file into a js object, then access each pattern array directly rather than manually copying each pattern to a new variable one tile at a time.

**Comment by nojronatron** [comment]
That's a good idea! I'll update this PR with that update.

**Comment by nojronatron** [comment]
I refactored the test files to use object literals. Let me know if this was/not what you were talking about. Fill me in if not. In any case it was a good exercise. :smiley:

**Comment by schaferyan** [comment]
This is a little bit different from what I imagined, but it's definitely condensed and easier to read. If you wanted to take it a step further, I think you can safely pass instance.dauberedTiles directly instead of copying into to a new tileList with `foreach()` and `push()`. I'll see if I can recommend this as a change.

File: src/testLib/calcBingoV.test.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+const calcBingoV = require('../funcLib/CalcBingoV.js');
+const testArrays = require('./testData/verticalBingoPatterns.json');
+const patterns = [];
+
+testArrays.forEach((instance) => {
+  const tileList = [];
```
Comment (schaferyan) [inline src/testLib/calcBingoV.test.js:6]:
```suggestion

```

File: src/testLib/calcBingoV.test.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  instance.dauberedTiles.forEach(tile => tileList.push(tile));
```
Comment (schaferyan) [inline src/testLib/calcBingoV.test.js:7]:
```suggestion

```

File: src/testLib/calcBingoV.test.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  patterns.push(
+    {
+      pattern: instance.pattern,
+      dauberedTiles: tileList
```
Comment (schaferyan) [inline src/testLib/calcBingoV.test.js:11]:
```suggestion
      dauberedTiles: instance.dauberedTiles
```

**Review by schaferyan — CHANGES_REQUESTED** [review]
If I did this correctly, you should see a suggested change in the calcBingoV.test.js that demonstrates my meaning

**Comment by nojronatron** [comment]
Updated code per Ryan's suggestions. Tests running and passing.

**Review by schaferyan — APPROVED** [review]
Looks good!

**Pull request merged**